helper/src/fs.rs

```diff
@@ -41,23 +41,39 @@ pub const CUPRATE_DIR: &str = {
     }
 };
 
+/// Cuprate's database filename.
+///
+/// This is the filename for Cuprate's database, used in [`cuprate_database_file`].
+///
+/// | OS      | PATH                                                                 |
+/// |---------|----------------------------------------------------------------------|
+/// | Windows | `C:\Users\Alice\AppData\Roaming\Cuprate\database\data.mdb`           |
+/// | macOS   | `/Users/Alice/Library/Application Support/Cuprate/database/data.mdb` |
+/// | Linux   | `/home/alice/.local/share/cuprate/database/data.mdb`                 |
+///
+/// # Monero Equivalent
+/// `data.mdb`
+pub const CUPRATE_DATABASE_FILE: &str = "data.mdb"; // INVARIANT: must be changed in `impl_path_oncelock_and_fn` as well
```

Comment:
this is LMDB's file name, I think we should just specify the dir and allow backends to  set file names

Reply:
Right, it needs to be `cfg_if`'ed based on the backend.

We need the direct PATH to the file because of `Env::disk_size_bytes()`. Do we need this though? Or access to the file in any future function?

Reply:
The Env config will hold the file name to the DB, so users would just specify the dir and the backend would add the file name and expose a function `path_to_file` or something, I can't see any part of Cuprate needing the file except for the DB threads.

So this constant should be in the back-end not helper.

Reply:
Ok, it's now defined in `database/` depending on the backend, and accessible via:
1. `Env::config(&self) -> &Config`
2. `Config::database_file(&self) -> &Path`